go/store/types/decimal.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"context"
+
+	"github.com/shopspring/decimal"
+
+	"github.com/liquidata-inc/dolt/go/store/hash"
+)
+
+type Decimal decimal.Decimal
+
+func (v Decimal) Value(ctx context.Context) (Value, error) {
+	return v, nil
+}
+
+func (v Decimal) Equals(other Value) bool {
+	v2, ok := other.(Decimal)
+	if !ok {
+		return false
+	}
+
+	return decimal.Decimal(v).Equal(decimal.Decimal(v2))
+}
+
+func (v Decimal) Less(nbf *NomsBinFormat, other LesserValuable) (bool, error) {
+	if v2, ok := other.(Decimal); ok {
+		return decimal.Decimal(v).LessThan(decimal.Decimal(v2)), nil
+	}
+	return DecimalKind < other.Kind(), nil
+}
+
+func (v Decimal) Hash(nbf *NomsBinFormat) (hash.Hash, error) {
+	return getHash(v, nbf)
+}
+
+func (v Decimal) isPrimitive() bool {
+	return true
+}
+
+func (v Decimal) WalkValues(ctx context.Context, cb ValueCallback) error {
+	return nil
+}
+
+func (v Decimal) WalkRefs(nbf *NomsBinFormat, cb RefCallback) error {
+	return nil
+}
+
+func (v Decimal) typeOf() (*Type, error) {
+	return PrimitiveTypeMap[DecimalKind], nil
+}
+
+func (v Decimal) Kind() NomsKind {
+	return DecimalKind
+}
+
+func (v Decimal) valueReadWriter() ValueReadWriter {
+	return nil
+}
+
+func (v Decimal) writeTo(w nomsWriter, nbf *NomsBinFormat) error {
+	encodedDecimal, err := decimal.Decimal(v).GobEncode()
```

Comment:
This dependency means that if shopspring ever makes a backwards incompatible change to their binary encoding, things break. I don't think this is super likely, but after you merge this you should add a bats test that will fail if it ever happens.